cmd/juju/application/deploy.go

```diff
@@ -720,7 +720,16 @@ func (c *DeployCommand) Init(args []string) error {
 		// do a late validation at Run().
 		c.unknownModel = true
 	}
-	if c.channelStr != "" {
+	if c.channelStr == "" {
+		if c.Revision != -1 {
+			// Tell the user they need to specify a channel
+			return errors.BadRequestf(
+				"If a revision is specified, a channel must also be specified for store\n" +
+					"charms and bundles. Do this using the --channel option, e.g.\n" +
+					"    juju deploy <charm> --revision=5 --channel=stable",
+			)
+		}
+	} else {
```

Comment:
Break out as it's own if statement as the previous one has a return in it.

Reply:
Here is a few points/questions that can help make code cleaner:
- Exit early.
- Prefer long functions over deep functions.
- Would inverting the if condition make the function simpler (i.e. longer over deeper)?
- Can you make the function shallower (less nested scopes) by moving a block to another function?